tests/platform_tests/test_intf_fec.py

```diff
@@ -147,3 +109,77 @@ def test_verify_fec_stats_counters(duthosts, enum_rand_one_per_hwsku_frontend_ho
         if fec_symbol_err_int > fec_corr_int:
             pytest.fail("FEC symbol errors:{} are higher than FEC correctable errors:{} for interface {}"
                         .format(intf_name, fec_symbol_err_int, fec_corr_int))
+
+
+def get_fec_histogram(duthost, intf_name):
+    """
+    @Summary: Fetch FEC histogram for a given interface.
+    """
+    try:
+        logging.info("Get output of 'show interfaces counters fec-histogram {}'".format(intf_name))
+        fec_hist = duthost.show_and_parse("show interfaces counters fec-histogram {}".format(intf_name))
+    except Exception as e:
+        logging.error("Failed to execute 'show interfaces counters fec-histogram {}': {}".format(intf_name, e))
+        pytest.fail("Command 'show interfaces counters fec-histogram {}' not found \
+                or failed: {}".format(intf_name, str(e)))
+        return None
+
+    if not fec_hist:
+        pytest.fail("No FEC histogram data found for interface {}".format(intf_name))
+
+    logging.info("FEC histogram for interface {}: {}".format(intf_name, fec_hist))
+    return fec_hist
+
+
+def collect_fec_histogram_samples(duthost, intf_name, num_samples=3, interval=10):
+    """
+    @Summary: Collect FEC histogram samples over a period of time.
+    """
+    fec_hist_samples = []
+    for sample_index in range(num_samples):
+        fec_hist = get_fec_histogram(duthost, intf_name)
+        fec_hist_samples.append(fec_hist)
+
+        # Log bin values for each sample
+        for bin_index in range(16):
+            bin_label = fec_hist[bin_index].get('symbol errors per codeword')
+            bin_value = fec_hist[bin_index].get('codewords')
+            logging.info("Sample {} - Interface {}: {} -> {}"
+                         .format(sample_index + 1, intf_name, bin_label, bin_value))
+
+        if sample_index < num_samples - 1:
+            time.sleep(interval)
+
+    return fec_hist_samples
+
+
+def validate_fec_histogram(fec_hist_samples, intf_name, critical_bins=range(7, 16)):
+    """
+    @Summary: Validate FEC histogram bins to ensure no increasing errors.
+    """
+    for bin_index in critical_bins:
+        previous_value = int(fec_hist_samples[0][bin_index].get('codewords', 0))
+        current_value = int(fec_hist_samples[2][bin_index].get('codewords', 0))
+
```

Comment:
@vvolam what is current value vs prev values? I see you are taking 3 samples but only indexing [0] and [2]...what about[1]?

Reply:
If there is a difference between [0] and [1] or [1] and [2], it will be captured when checking with [0] and [2]. Do you want to fail the test if difference is found in [1] sample itself?

Reply:
@prgeor updated the PR according to our offline discussion. Fail the test if any errors found in bins > 7 and also check for 3 times in the intervals of 10. Thanks!